charts/nutanix-csi-snapshot/templates/validating-webhook.yaml

```diff
@@ -0,0 +1,52 @@
+# Check if the TLS secret already exists and initialize variables for later use at the top level
+{{- $secret := lookup "v1" "Secret" .Release.Namespace .Values.tls.secretName }}
+{{ $ca := "" }}
+{{ $key := "" }}
+{{ $crt := "" }}
+---
+{{- if (eq .Values.tls.source "generate") }}
+  {{- if and $secret (not .Values.tls.renew) }}
+    {{- $ca = get $secret.data "ca.crt" }}
+    {{- $key = get $secret.data "tls.key" }}
+    {{- $crt = get $secret.data "tls.crt" }}
+  {{- else }}
+    {{- $serviceName := (printf "%s.%s.svc" "snapshot-validation-service" .Release.Namespace)}}
+    {{- $cert := genSelfSignedCert $serviceName nil (list $serviceName) 3650 }}
```

Comment:
Where is genSelfSignedCert defined?

Reply:
Found it https://helm.sh/docs/chart_template_guide/function_list/#genselfsignedcert :) Had no idea helm shipped with such an advanced (and convenient) function :)

Reply:
Exactly the built-in function library is huge